From 38fc7ef885bf0eeaceb6490c774740cc94770dac Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Fri, 15 May 2020 10:08:24 -0400
Subject: [PATCH] local-display-factory: always force login screen to VT 1

These days we always want the login screen on VT 1, even
when it's created by user switching.

Unfortunately, since commit f843233ad the login screen
won't naturally pick VT 1 when user switching.

This commit forces it to make the right choice.
---
 daemon/gdm-local-display-factory.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/daemon/gdm-local-display-factory.c b/daemon/gdm-local-display-factory.c
index da1093bb..8709edd0 100644
--- a/daemon/gdm-local-display-factory.c
+++ b/daemon/gdm-local-display-factory.c
@@ -224,6 +224,7 @@ gdm_local_display_factory_create_transient_display (GdmLocalDisplayFactory *fact
 {
         gboolean         ret;
         GdmDisplay      *display = NULL;
+        gboolean         is_initial = FALSE;
 
         g_return_val_if_fail (GDM_IS_LOCAL_DISPLAY_FACTORY (factory), FALSE);
 
@@ -235,6 +236,7 @@ gdm_local_display_factory_create_transient_display (GdmLocalDisplayFactory *fact
         display = gdm_local_display_new ();
         if (gdm_local_display_factory_use_wayland ())
                 g_object_set (G_OBJECT (display), "session-type", "wayland", NULL);
+        is_initial = TRUE;
 #else
         if (display == NULL) {
                 guint32 num;
@@ -248,6 +250,7 @@ gdm_local_display_factory_create_transient_display (GdmLocalDisplayFactory *fact
         g_object_set (display,
                       "seat-id", "seat0",
                       "allow-timed-login", FALSE,
+                      "is-initial", is_initial,
                       NULL);
 
         store_display (factory, display);
-- 
2.26.2

